### PR TITLE
Update `add_prop_ci()` to `add_ci()` and include CIs for continuous variables.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update NEWS.md with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parantheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/add_ci.R [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+#' @param statistic Formula indicating how the confidence interval will be placed.
+#' Default is `list(all_categorical() ~ "{conf.low}%, {conf.high}%", all_continuous() ~ "{conf.low}, {conf.high}")`
+#' @param method Confidence interval method. Default is
+#' `list(all_categorical() ~ "wilson", all_continuous() ~ "t.test")`.
+#' Must be one of
+#' `c("wilson", "wilson.no.correct", "exact", "asymptotic")` for categorical
+#' variables, and `"t.test"` for continuous variables.
+#' See details below.
+#' @param conf.level Confidence level. Default is `0.95`
+#' @param ci_fun Function to style upper and lower bound of confidence
```
Comment (larmarange) [inline R/add_ci.R:15]: Could `ci_fun` be confusing, i.e. corresponding to the function to compute a CI and not a function to format/style the values?

To be consistent with other gtsummary functions, calling this parameter `style_fun`?

**Review by larmarange — COMMENTED** [review]
Hey, The code and the functionality are good.

Maybe just a few remarks regarding improvement of documentation. 

I'm not sure about the name of the argument `ci_fun`. For me it would be expected a function to compute rather than style the CIs. Maybe a name like `style_fns` would be more consistent (we will understood that we exemect a `style_*` function). 

Other thing, in `add_stat()`, functions is abbreviated in fns not fun. Maybe it would be nice to improve consistency between functions.

Best regards

File: R/add_ci.R [hunk 2]
```diff
@@ -0,0 +1,283 @@
+#' Add Proportion CIs
+#'
+#' Add a new column with the confidence intervals for proportions.
+#'
+#' @param x A `tbl_summary` object
+#' @param statistic Formula indicating how the confidence interval will be placed.
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_ci.R:6]: updated!

File: R/add_ci.R [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+#' @param conf.level Confidence level. Default is `0.95`
+#' @param ci_fun Function to style upper and lower bound of confidence
+#' interval. Default is
+#' `list(all_categorical() ~ purrr::partial(style_sigfig, scale =  100), all_continuous() ~ style_sigfig)`.
+#' @param ... Not used
+#' @inheritParams tbl_summary
+#'
+#' @section method argument:
+#' Methods `c("wilson", "wilson.no.correct")` are calculated with `prop.test()`.
+#' The default method, `"wilson"`, includes the Yates continuity correction.
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_ci.R:23]: updated!

File: R/add_ci.R [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+#' Methods `c("exact", "asymptotic")` are calculated with `Hmisc::binconf(method=)`.
+#'
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_ci.R:25]: added!

File: R/add_ci.R (hunk 1 above)
Comment (ddsjoberg) [reply R/add_ci.R:15]: updated to style_fun!~

File: R/add_ci.R [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+#' @return gtsummary table
+#' @rdname add_ci
+#' @export
+#'
+#' @family tbl_summary tools
+#' @examples
+#' # Example 1 ----------------------------------
+#' add_ci_ex1 <-
+#'   trial %>%
+#'   select(response, trt) %>%
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_ci.R:35]: added

**Review by larmarange — APPROVED** [review]
Thanks

**Comment by ddsjoberg** [comment]
> Thanks

Thank you @larmarange !

I think it may be worth doing a survey of the package, and making items more consistent....but I can't gather the strength for it at the moment 😆

**Pull request merged**